HopsanGenerator/src/generators/HopsanExeGenerator.cpp

```diff
@@ -0,0 +1,292 @@
+        return false;
+    }
+
+    QStringList objectFiles;
+    objectFiles << "exe_main.o" << "exe_utilities.o";
+    for(const QString& extraSrc : mExtraSourceFiles) {
+        QFileInfo extraObjFile(extraSrc);
+        objectFiles << extraObjFile.baseName()+".o";
+    }
+    for(const QString &srcFile : srcFiles) {
+        QFileInfo fi(srcFile);
+        objectFiles << fi.baseName()+".o";
+    }
+
+    if(!assertFilesExist(buildPath, objectFiles)) {
+        return false;
+    }
+
+    printMessage("------------------------------------------------------------------------");
+    printMessage("Linking");
+    printMessage("------------------------------------------------------------------------");
+
+    bool linkingOK=false;
+
+    QString outputExecutableFile;
+#ifdef _WIN32
+    outputExecutableFile = QString("%1/%2.exe").arg(buildPath).arg(modelName);
+#elif __linux__
+    outputExecutableFile = QString("%1/%2").arg(buildPath).arg(modelName);
+#endif
+    //! @todo Add OSX support
+    // Create output directory before linking
+    QDir systemRootDir("");
+    systemRootDir.mkpath(QFileInfo(outputExecutableFile).absolutePath());
+
+#ifdef _WIN32
+    QFile linkBatchFile;
+    linkBatchFile.setFileName(buildPath + "/link.bat");
+    if(!linkBatchFile.open(QIODevice::WriteOnly | QIODevice::Text))
+    {
+        printErrorMessage("Failed to open link.bat for writing.");
+        return false;
+    }
+    //Write the compilation script file
+    QTextStream linkBatchStream(&linkBatchFile);
+    linkBatchStream << "@echo off\n";
+    linkBatchStream << "PATH=" << mCompilerSelection.path << ";%PATH%\n";
+    linkBatchStream << "@echo on\n";
+    linkBatchStream << "g++ -w -static -static-libgcc";
+    Q_FOREACH(const QString &objFile, objectFiles)
+    {
+        linkBatchStream << " " << objFile;
+    }
+    linkBatchStream << " -o "+outputExecutableFile+"\n";
+    linkBatchFile.close();
+
+    linkingOK = callProcess("cmd.exe", QStringList() << "/c" << "cd /d " + buildPath + " & link.bat");
+
+#else
+    QFile linkBatchFile;
+    linkBatchFile.setFileName(buildPath + "/link.sh");
+    if(!linkBatchFile.open(QIODevice::WriteOnly | QIODevice::Text))
+    {
+        printErrorMessage("Failed to open link.sh for writing.");
+        return false;
+    }
+    //Write the compilation script file
+    QTextStream linkBatchStream(&linkBatchFile);
+    linkBatchStream << mCompilerSelection.path+"g++ -pthread -w";
+    Q_FOREACH(const QString &objFile, objectFiles)
+    {
+        linkBatchStream << " " << objFile;
+    }
+    linkBatchStream << " -o "+outputExecutableFile+" -ldl\n";
+    linkBatchFile.close();
+
+    linkingOK = callProcess("/bin/sh", QStringList() << "link.sh", buildPath);
+
+#endif
+
+    if (!linkingOK) {
+        printErrorMessage("Failed to link exported code for executable model.");
+        return false;
+    }
+
+    QDir outputDir = QFileInfo(outputExecutableFile).absoluteDir();
+    outputDir.cdUp();
+#ifdef WIN32
```

Comment:
this should be _WIN32, WIN32 without leading _ does not always exist

Reply:
Ops, my bad. It is fixed.